paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSource.java

```diff
@@ -53,11 +69,42 @@ public SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint> restoreEnu
             PendingSplitsCheckpoint checkpoint) {
         Collection<FileStoreSourceSplit> splits =
                 checkpoint == null ? getSplits() : checkpoint.splits();
-        return new StaticFileStoreSplitEnumerator(context, null, splits, splitBatchSize);
+        SplitAssigner splitAssigner =
+                createSplitAssigner(context, splitBatchSize, splitAssignMode, splits);
+        return new StaticFileStoreSplitEnumerator(context, null, splitAssigner);
     }
 
     private List<FileStoreSourceSplit> getSplits() {
         FileStoreSourceSplitGenerator splitGenerator = new FileStoreSourceSplitGenerator();
         return splitGenerator.createSplits(readBuilder.newScan().plan());
     }
+
+    public static SplitAssigner createSplitAssigner(
```

Comment:
move this method into `FairSplitAssigner`.

Reply:
Sorry, I don't quite understand why this method needs to be moved to `FairSplitAssigner`.  Maybe you mean [`createSplitAssignment`](https://github.com/apache/incubator-paimon/pull/1301/files#diff-a12ee210f24e59beb65f3d656f97ac8aa4d16dcce95b2491ce61ae83ce734a1aR99)?

Reply:
Yes, this is a mistake..